views/dashboard/testimonials/index.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState, useMemo } from 'react'
+import { TestimonialsProps } from '@/utils/types/redux'
+import { useSelector } from 'react-redux'
+import TestimonialCard from '@/components/dashboard/cards/testimonial'
+
+import { testimonialFilterMap, testimonialSortMap } from '@/utils/constants'
+import SelectField from '@/components/form/fields/SelectField'
+import RadioFieldGroup from '@/components/form/fields/RadioFieldGroup'
+import {
+    StyledTestimonialGrid,
+    StyledSortAndFilter
+} from './StyledTestimonials'
+
+const TestimonialWidget = () => {
+    const { reviews } = useSelector((state: any) => state.testimonials)
+    const [displaySort, setDisplaySort] = useState<string>('')
+    const [displayFilter, setDisplayFilter] = useState<string>('SHOW_ALL')
+
+    const handleDisplaySort = (sortType: string) => setDisplaySort(sortType)
+    const handleDisplayFilter = (filterType: string) =>
+        setDisplayFilter(filterType)
+    const handleRenderReviews = () => {
+        const filtered = testimonialFilterMap
+            .get(displayFilter)
+            ?.run([...reviews])
+        const sorted = testimonialSortMap.get(displaySort)?.run(filtered)
+        return sorted
+    }
+    const displayReviews = useMemo(
+        () => handleRenderReviews(),
+        [displayFilter, displaySort, reviews]
+    )
+
+    return (
+        <>
```

Comment:
Seeing this feels off to me. This should be wrapped in a StyledTestimonialsWidget or something. To apply the base styles to the entire page. Double check why that's not done here

Reply:
just a small fragment of big things to come

Reply:
LMAO